tests/vm_test.go

```diff
@@ -528,11 +528,16 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			vm := createRunningVM(virtClient, vmi)
 
 			By("Verifying that the status toggles between ErrImagePull and ImagePullBackOff")
-			const times = 2
-			for i := 0; i < times; i++ {
-				Eventually(ThisVM(vm), 300*time.Second, 1*time.Second).Should(HavePrintableStatus(v1.VirtualMachineStatusErrImagePull))
-				Eventually(ThisVM(vm), 300*time.Second, 1*time.Second).Should(HavePrintableStatus(v1.VirtualMachineStatusImagePullBackOff))
+			maxImgPullRetries := 2 // max number of image pull (re)tries
+			expectedStates := []v1.VirtualMachinePrintableStatus{}
+			for range maxImgPullRetries {
+				// each image pull (re)try should take the VM
+				// from status ErrImagePull to ImagePullBackOff
+				expectedStates = append(expectedStates, v1.VirtualMachineStatusErrImagePull)
+				expectedStates = append(expectedStates, v1.VirtualMachineStatusImagePullBackOff)
```

Comment:
You can fold this

```suggestion
			expectedStates := []v1.VirtualMachinePrintableStatus{
				v1.VirtualMachineStatusErrImagePull,
				v1.VirtualMachineStatusImagePullBackOff,
			}
```

Reply:
I would like to retain the logic wherein the test checks the expected state transitions over 2 image pull retries. Will update accordingly with your suggestion!

Reply:
Applied the suggested change.